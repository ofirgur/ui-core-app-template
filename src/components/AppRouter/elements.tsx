import React, { Suspense } from 'react';

export const getAppElement = () => {
  const Component = React.lazy(
    () => import(/* webpackChunkName: "app-element" */ 'components/App')
  );

  return (
    <Suspense fallback={<>...</>}>
      <Component />
    </Suspense>
  );
};

export const getDashboardElement = () => {
  const Component = React.lazy(
    () => import(/* webpackChunkName: "dashboard-element" */ 'pages/dashboard')
  );

  return (
    <Suspense fallback={<>...</>}>
      <Component />
    </Suspense>
  );
};

export const getGarmentsElement = () => {
  const Component = React.lazy(
    () => import(/* webpackChunkName: "garments-element" */ 'pages/garments')
  );

  return (
    <Suspense fallback={<>...</>}>
      <Component />
    </Suspense>
  );
};

export const getCollectionsElement = () => {
  const Component = React.lazy(
    () =>
      import(/* webpackChunkName: "collections-element" */ 'pages/collections')
  );

  return (
    <Suspense fallback={<>...</>}>
      <Component />
    </Suspense>
  );
};

export const getNotFoundElement = () => {
  const Component = React.lazy(
    () => import(/* webpackChunkName: "not-found-element" */ 'pages/notFound')
  );

  return (
    <Suspense fallback={<>...</>}>
      <Component />
    </Suspense>
  );
};
