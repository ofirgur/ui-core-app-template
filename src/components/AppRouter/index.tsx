import * as React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Dashboard from 'pages/dashboard';
import Garments from 'pages/garments';
import Collections from 'pages/collections';
import App from '../App';

const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<App />}>
          <Route index element={<Dashboard />} />
          <Route path="garments" element={<Garments />} />
          <Route path="collections" element={<Collections />} />
        </Route>
      </Routes>
    </BrowserRouter>
  );
};

export default AppRouter;
