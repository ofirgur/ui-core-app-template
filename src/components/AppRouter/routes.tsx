export default {
  dashboard: {
    pathname: '/',
    title: 'Dashboard',
  },
  garments: {
    pathname: 'garments',
    title: 'Garments',
  },
  collection: {
    pathname: 'collection',
    title: 'Collection',
  },
  notFound: {
    pathname: '*',
    title: 'Not Found',
  },
};
