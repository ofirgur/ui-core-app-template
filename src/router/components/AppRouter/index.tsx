import React from 'react';
import { BrowserRouter, Routes, Route, Navigate } from 'react-router-dom';
import routes from '../../routes';
import * as ELEMENTS from '../../elements';

const {
  home,
  manageproducts,
  createajob,
  manageavatars,
  managegarments,
  status,
  notfound,
} = routes;

const AppRouter = () => {
  const getPathname = (path: string) => `${path}/*`;

  return (
    <BrowserRouter>
      <Routes>
        <Route path={home.path} element={ELEMENTS.getAppElement()}>
          <Route
            index
            element={<Navigate to={manageproducts.path} replace />}
          />
          <Route
            path={getPathname(manageproducts.path)}
            element={ELEMENTS.getManageProductsElement()}
          />
          <Route
            path={getPathname(createajob.path)}
            element={ELEMENTS.getCreateAJobElement()}
          />
          <Route
            path={getPathname(manageavatars.path)}
            element={ELEMENTS.getManageAvatarsElement()}
          />
          <Route
            path={getPathname(managegarments.path)}
            element={ELEMENTS.getManageGarmentsElement()}
          />
          <Route
            path={getPathname(status.path)}
            element={ELEMENTS.getStatusElement()}
          />
          <Route path={notfound.path} element={ELEMENTS.getNotFoundElement()} />
        </Route>
      </Routes>
    </BrowserRouter>
  );
};

export default AppRouter;
