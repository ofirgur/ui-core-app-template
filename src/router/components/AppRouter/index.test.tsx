import * as React from 'react';
import { render } from '@testing-library/react';
import AppRouter from '.';

describe('AppRouter', () => {
  it('should render correctly', () => {
    const { container } = render(<AppRouter />);
    expect(container).toMatchSnapshot();
  });
});
