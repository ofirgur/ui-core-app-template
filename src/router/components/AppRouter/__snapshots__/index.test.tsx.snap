// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter should render correctly 1`] = `
<div>
  ...
</div>
`;
